src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -577,6 +583,13 @@ protected virtual void BuildOptions<T>(IRocksDbConfig dbConfig, Options<T> optio
             }
         }
 
+        if (mergeOperator is not null)
+        {
+            options.SetMergeOperator(new MergeOperatorAdapter(mergeOperator));
+            lock (_doNotGcOptionsLocker)
+                _doNotGcOptions.Add(options);
```

Comment:
I think `_doNotGcOptionsLocker` is redundant, you can just do:


```suggestion
            lock (_doNotGcOptions)
                _doNotGcOptions.Add(options);
```